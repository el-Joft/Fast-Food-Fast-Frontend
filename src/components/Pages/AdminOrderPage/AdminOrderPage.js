import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getAdminOrders, deleteAnOrder } from '../../../actions/orderAction';

class AdminOrderPage extends Component {
  componentWillMount() {
    const token = localStorage.getItem('userToken');
    this.props.getAdminOrders(token);
  }

  deleteOrder = (e, id) => {
    e.preventDefault();
    const token = localStorage.getItem('userToken');
    const { history } = this.props;
    this.props.deleteAnOrder(id, token, history);
  };

  renderCard = () => this.props.orders.map((item, i) => (
    <tbody key={i}>
      <tr>
        <td>{item.name}</td>
        <td>{item.description}</td>
        <td>{item.price}</td>
        <td>{item.quantity}</td>
        <td>{item.firstname}</td>
        <td>{item.totalprice}</td>
        <td>
          <div className="">
            <button onClick={e => this.deleteOrder(e, item.id)} className='delete-order' type='submit'>Delete</button>
          </div>
        </td>
      </tr>
    </tbody>
  ))

  render() {
    return (
      <div>
        {
          this.props.success.status
            ? <div className='success'>Sucessfully deleted an order</div>
            : null
        }

        <table className="admin-table">

          <thead className="table-head">
          <tr>
            <th>Item</th>
            <th>Description</th>
            <th>
Price (
              <strike>N</strike>
)
            </th>
            <th>Qty</th>
            <th>User</th>
            <th>
Total (
              <strike>N</strike>
)
            </th>
            <th>Actions</th>
          </tr>
        </thead>
          {
          this.props.orders === null
            ? null
            : this.renderCard()}
        </table>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  auth: state.auth,
  orders: state.orders.orders,
  success: state.success,
  errors: state.errors,
});

export default connect(mapStateToProps, { getAdminOrders, deleteAnOrder })(AdminOrderPage);
