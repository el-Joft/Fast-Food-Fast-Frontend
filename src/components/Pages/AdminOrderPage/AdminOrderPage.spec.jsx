/* eslint-disable react/react-in-jsx-scope */
import { shallow } from 'enzyme';
import { AdminOrderPage } from './AdminOrderPage';

describe('rendering', () => {
  let wrapper;
  let props;
  let mockData;
  beforeEach(() => {
    mockData = {
      auth: {
        isAuthenticated: true,
        loading: null,
        user: { id: 1, iat: 1549439277, exp: 1549525677 },
      },
      errors: {
        isAuthenticated: false,
        loading: null,
        email: '',
        password: '',
        message: '',
      },
      loader: {
        loading: false,
      },
    };
    props = {
      getAdminOrders: jest.fn(),
      comment: '',
      success: {
        status: true,
      },
      orders: [],
      auth: {
        user: {
          id: 1,
        },
      },
    };
  });

  it('invokes `componentWillMount` when mounted', () => {
    const spy = jest.spyOn(AdminOrderPage.prototype, 'componentWillMount');
    wrapper = shallow(<AdminOrderPage {...props} />);
    wrapper.instance().componentWillMount();
    expect(spy).toHaveBeenCalled();
  });

  it('should render the component', () => {
    wrapper = shallow(<AdminOrderPage {...props} />);
    expect(wrapper.length).toBe(1);
  });

  // it('should test the onSubmit function', () => {
  //   const fakeEvent = { preventDefault: () => console.log('preventDefault') };
  //   wrapper = shallow(<AdminOrderPage {...props} />);
  //   const loginForm = wrapper.find('button');
  //   loginForm.simulate('click', fakeEvent);
  //   const logoutUser = jest.fn(mockData);
  //   const promise = new Promise((resolve) => {
  //     resolve(wrapper.instance().onClick);
  //   });
  //   promise.then(() => expect(logoutUser).toHaveBeenCalledTimes(1));
  // });


  it('should render the component', () => {
    wrapper = shallow(<AdminOrderPage {...props} />);
    const td = wrapper.find('table');
    expect(td.length).toBe(1);
  });

  it('should render the component', () => {
    wrapper = shallow(<AdminOrderPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
