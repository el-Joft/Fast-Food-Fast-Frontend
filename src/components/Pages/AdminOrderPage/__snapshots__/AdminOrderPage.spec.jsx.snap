// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminOrderPage
    auth={
      Object {
        "user": Object {
          "id": 1,
        },
      }
    }
    comment=""
    getAdminOrders={
      [MockFunction] {
        "calls": Array [
          Array [
            null,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    orders={Array []}
    success={
      Object {
        "status": true,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="success"
        >
          Sucessfully deleted an order
        </div>,
        <table
          className="admin-table"
        >
          <thead
            className="table-head"
          >
            <tr>
              <th>
                Item
              </th>
              <th>
                Description
              </th>
              <th>
                Price (
                <strike>
                  N
                </strike>
                )
              </th>
              <th>
                Qty
              </th>
              <th>
                User
              </th>
              <th>
                Total (
                <strike>
                  N
                </strike>
                )
              </th>
              <th>
                Actions
              </th>
            </tr>
          </thead>
        </table>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sucessfully deleted an order",
          "className": "success",
        },
        "ref": null,
        "rendered": "Sucessfully deleted an order",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead
              className="table-head"
            >
              <tr>
                <th>
                  Item
                </th>
                <th>
                  Description
                </th>
                <th>
                  Price (
                  <strike>
                    N
                  </strike>
                  )
                </th>
                <th>
                  Qty
                </th>
                <th>
                  User
                </th>
                <th>
                  Total (
                  <strike>
                    N
                  </strike>
                  )
                </th>
                <th>
                  Actions
                </th>
              </tr>
            </thead>,
            Array [],
          ],
          "className": "admin-table",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th>
                  Item
                </th>
                <th>
                  Description
                </th>
                <th>
                  Price (
                  <strike>
                    N
                  </strike>
                  )
                </th>
                <th>
                  Qty
                </th>
                <th>
                  User
                </th>
                <th>
                  Total (
                  <strike>
                    N
                  </strike>
                  )
                </th>
                <th>
                  Actions
                </th>
              </tr>,
              "className": "table-head",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th>
                    Item
                  </th>,
                  <th>
                    Description
                  </th>,
                  <th>
                    Price (
                    <strike>
                      N
                    </strike>
                    )
                  </th>,
                  <th>
                    Qty
                  </th>,
                  <th>
                    User
                  </th>,
                  <th>
                    Total (
                    <strike>
                      N
                    </strike>
                    )
                  </th>,
                  <th>
                    Actions
                  </th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Item",
                  },
                  "ref": null,
                  "rendered": "Item",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Description",
                  },
                  "ref": null,
                  "rendered": "Description",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Price (",
                      <strike>
                        N
                      </strike>,
                      ")",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Price (",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "N",
                      },
                      "ref": null,
                      "rendered": "N",
                      "type": "strike",
                    },
                    ")",
                  ],
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Qty",
                  },
                  "ref": null,
                  "rendered": "Qty",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "User",
                  },
                  "ref": null,
                  "rendered": "User",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Total (",
                      <strike>
                        N
                      </strike>,
                      ")",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Total (",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "N",
                      },
                      "ref": null,
                      "rendered": "N",
                      "type": "strike",
                    },
                    ")",
                  ],
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Actions",
                  },
                  "ref": null,
                  "rendered": "Actions",
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            "type": "thead",
          },
        ],
        "type": "table",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="success"
          >
            Sucessfully deleted an order
          </div>,
          <table
            className="admin-table"
          >
            <thead
              className="table-head"
            >
              <tr>
                <th>
                  Item
                </th>
                <th>
                  Description
                </th>
                <th>
                  Price (
                  <strike>
                    N
                  </strike>
                  )
                </th>
                <th>
                  Qty
                </th>
                <th>
                  User
                </th>
                <th>
                  Total (
                  <strike>
                    N
                  </strike>
                  )
                </th>
                <th>
                  Actions
                </th>
              </tr>
            </thead>
          </table>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sucessfully deleted an order",
            "className": "success",
          },
          "ref": null,
          "rendered": "Sucessfully deleted an order",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <thead
                className="table-head"
              >
                <tr>
                  <th>
                    Item
                  </th>
                  <th>
                    Description
                  </th>
                  <th>
                    Price (
                    <strike>
                      N
                    </strike>
                    )
                  </th>
                  <th>
                    Qty
                  </th>
                  <th>
                    User
                  </th>
                  <th>
                    Total (
                    <strike>
                      N
                    </strike>
                    )
                  </th>
                  <th>
                    Actions
                  </th>
                </tr>
              </thead>,
              Array [],
            ],
            "className": "admin-table",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <th>
                    Item
                  </th>
                  <th>
                    Description
                  </th>
                  <th>
                    Price (
                    <strike>
                      N
                    </strike>
                    )
                  </th>
                  <th>
                    Qty
                  </th>
                  <th>
                    User
                  </th>
                  <th>
                    Total (
                    <strike>
                      N
                    </strike>
                    )
                  </th>
                  <th>
                    Actions
                  </th>
                </tr>,
                "className": "table-head",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th>
                      Item
                    </th>,
                    <th>
                      Description
                    </th>,
                    <th>
                      Price (
                      <strike>
                        N
                      </strike>
                      )
                    </th>,
                    <th>
                      Qty
                    </th>,
                    <th>
                      User
                    </th>,
                    <th>
                      Total (
                      <strike>
                        N
                      </strike>
                      )
                    </th>,
                    <th>
                      Actions
                    </th>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Item",
                    },
                    "ref": null,
                    "rendered": "Item",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Description",
                    },
                    "ref": null,
                    "rendered": "Description",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Price (",
                        <strike>
                          N
                        </strike>,
                        ")",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Price (",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "N",
                        },
                        "ref": null,
                        "rendered": "N",
                        "type": "strike",
                      },
                      ")",
                    ],
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Qty",
                    },
                    "ref": null,
                    "rendered": "Qty",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "User",
                    },
                    "ref": null,
                    "rendered": "User",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Total (",
                        <strike>
                          N
                        </strike>,
                        ")",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Total (",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "N",
                        },
                        "ref": null,
                        "rendered": "N",
                        "type": "strike",
                      },
                      ")",
                    ],
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Actions",
                    },
                    "ref": null,
                    "rendered": "Actions",
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
              "type": "thead",
            },
          ],
          "type": "table",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
